libexec/rtld-elf/aarch64/rtld_machdep.h

```diff
@@ -151,13 +151,29 @@ make_data_cap(const Elf_Sym *def, const struct Struct_Obj_Entry *defobj)
 
 /* TODO: Per-function captable/PLT/FNDESC support */
 #ifdef RTLD_SANDBOX
-#define make_rtld_function_pointer(target_func)	(tramp_pgs_append((target_func), NULL, NULL))
-
-#define call_init_array_pointer(obj, target)				\
-	(((InitArrFunc)tramp_pgs_append((void *)(target).value, obj, NULL))(main_argc, main_argv, environ))
-
-#define call_fini_array_pointer(obj, target)				\
-	(((InitFunc)tramp_pgs_append((void *)(target).value, obj, NULL))())
+#define make_rtld_function_pointer(_target)				\
+	(tramp_intern(NULL, &(struct tramp_data) {			\
+		.target = _target,					\
+		.defobj = &obj_rtld,					\
+		.sig = (struct tramp_sig) { .valid = true,		\
+		    .reg_args = 0, .mem_args = false, .ret_args = NONE }\
```

Comment:
Yeah, this only works because `make_rtld_function_pointer` is currently only used with functions that take no arguments and return nothing. What would you suggest? Making `make_rtld_function_pointer` take the signature as well? Or call `tramp_intern` explicitly after a no-op `make_rtld_function_pointer`? I personally favour the latter.

Reply:
Should be consistent with make_function_pointer I guess, whatever ends up being the approach taken there